### PR TITLE
Don't realloc null terminated strings during utf8 encoding

### PR DESCRIPTION
**Pull request by MaxGraey**

### CODE REVIEW

File: std/assembly/string.ts [hunk 1]
```diff
@@ -661,12 +661,13 @@ export namespace String {
     export function encode(str: string, nullTerminated: bool = false): ArrayBuffer {
       var strOff = changetype<usize>(str);
       var strEnd = changetype<usize>(str) + <usize>changetype<BLOCK>(changetype<usize>(str) - BLOCK_OVERHEAD).rtSize;
-      var buf = __alloc(UTF8.byteLength(str, nullTerminated), idof<ArrayBuffer>());
+      var bufLen = <usize>UTF8.byteLength(str, nullTerminated);
+      var buf = __alloc(bufLen, idof<ArrayBuffer>());
+      var bufEnd = buf + bufLen - i32(nullTerminated);
```
Comment (dcodeIO) [inline std/assembly/string.ts]: Makes me wonder if we have `usize(nullTerminated)` in portable?

Comment (MaxGraey) [reply std/assembly/string.ts]: makes sense

Comment (dcodeIO) [reply std/assembly/string.ts]: Seems we don't. Should probably add this as an alias of `u32` to be correct.

**Comment by dcodeIO** [comment]
Thanks! :)

**Pull request merged**